projects/tuist/fixtures/app_with_spm_dependencies/Tuist/Dependencies.swift

```diff
@@ -2,22 +2,9 @@
 import ProjectDescription
 import ProjectDescriptionHelpers
 
-let packages: [Package] = [
-    .package(url: "https://github.com/Alamofire/Alamofire", .upToNextMajor(from: "5.8.0")),
-    .package(url: "https://github.com/facebook/facebook-ios-sdk", .upToNextMajor(from: "16.1.3")),
-    .package(url: "https://github.com/firebase/firebase-ios-sdk", .upToNextMajor(from: "10.15.0")),
-    .package(url: "https://github.com/pointfreeco/swift-composable-architecture", .upToNextMinor(from: "1.0.0")),
-    .package(url: "https://github.com/iterable/swift-sdk", .upToNextMajor(from: "6.4.15")),
-    .package(url: "https://github.com/Trendyol/ios-components", .revision("c9260bfe203a16a278eca5542c98455eece98aa4")),
-    .package(url: "https://github.com/stripe/stripe-ios", .upToNextMajor(from: "23.12.0")),
-    .local(path: "LocalSwiftPackage"),
-    .package(url: "https://github.com/groue/GRDB.swift", .upToNextMajor(from: "6.16.0")),
-    .local(path: "StringifyMacro"),
-]
-
 let dependencies = Dependencies(
     swiftPackageManager: .init(
```

Comment:
Long-term, I wonder if we should remove `Dependencies.swift` in favor of `Tuist/Config.swift` to cut down on the number of manifests. `Dependencies.swift` won't contain much now – primarily overrides for individual dependencies.

We could even assume a developer wants to use our SPM integration based on the existence of `Package.swift` in `Tuist/Dependencies`. Probably better to slate these changes for Tuist 4.0 – we should couple that with the removal of Carthage.

Maybe we can start envisioning what we want to cut in Tuist 4.0? cc @pepicrft

Reply:
I had dumped similar thoughts in another comment, so I'm moving them here:

---

What do you think about taking a different API approach here? Considering a future where SPM is the default for most projects, which I believe is very realistic, I don't see the need for a `Dependencies.swift` in the project. In that world, we could have a default, for example, a `Package.swift` next to the `Tuist` directory, and an option in the `Config.swift` to override the path.
The motivation to suggest that path as a default is that it'll usually match the root of the repository, and that'll make setting up dependency auto-updates like Dependabot more straightforward. Even though the directory is configurable there, their auto-setup logic will most likely do the lookup in the root to understand the type of package manager they are working with.

In preparation for that work, if we all agree is the way forward, I'd keep the current behavior but **add a warning** message that nudges developers to adopt the new pattern. That way, by the time we drop support for `Dependencies.swift`, it won't be a breaking change for most of the users.